HHS/Head-Start-TTADP
# [TTAHUB-2530] Remove fields from objective form in RTR

## Description of change
- Removed topics, resources, support type, objective status, courses, and files from RTR
- The "view goal" from the RTR is a little different, so I created a new template to stop the logic loops inside the create form from getting even crazier. To do that and keep things consistent, some bits of display were moved into reusable components
- Cleaned up the route on the backend for this page. We no longer care about "ObjectiveTopics", "ObectiveResources", etc, only their "ActivityReportObjectivesTopics" and the like counterparts. Moved it to Typescript to make refactors a little easier in the future.
- Removed an unused route on the backend
- Note that other PRs within this sprint will build on this work, altering the AR fields and cleaning up the database schema to remove unused tables.

## How to test
- Create a new goal on the RTR, with an objective. Confirm that the  topics, resources, support type, objective status, courses, and files fields aren't there.
- Edit that goal. If the objective isn't used on an activity report, the text should still be editable.
- Use that goal and objective on a report. Once the objective is on a report, the text is no longer editable.
- Approve the report and close the goal. Go the RTR and confirm that the view option is present. View the goal and confirm that it matches the design.
- Test the same cases with FEI goals and multirecipient reports.

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-2530


## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
